nncf/common/graph/graph.py

```diff
@@ -348,20 +348,52 @@ def get_input_edges(self, node: NNCFNode) -> List[NNCFGraphEdge]:
             edges.extend(self._get_edges(from_node, node))
         return sorted(edges, key=lambda x: x.input_port_id)
 
+    def get_input_edge_by_port_id(self, node: NNCFNode, port_id: int) -> NNCFGraphEdge:
```

Comment:
Could briefly explain why you added this function instead of using get_input_edges(node)[port_id]?

Reply:
We apply passes, such as `remove_shapeof_subgraphs()` and `filter_constant_nodes()`, to the graph that can invalidate the order of input edges. For this reason, it is safer to use the `get_input_edge_by_port_id()` method.